packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -86,7 +86,7 @@ export function extractConfigByNamespace(Component, dataset, namespace) {
         // Drop down to what we assume is a nested object
         // but check for object type on the next loop
         if (index < keyParts.length - 1) {
-          current = current[name] = current[name] ?? {}
+          current = current[name] = isObject(current[name]) ? current[name] : {}
```

Comment:
No issue with the change of logic, but the commit attracted my attention to this line. I think it's compact but a bit tricky to follow. Can it be split into the following (which if I understood wrong further highlights the need to separate the statements 😆 )?

```suggestion
current[name] = isObject(current[name]) ? current[name] : {}
current = current[name]
```

Reply:
@romaricpascal Found the perfect ESLint rule for you [`no-multi-assign`](https://eslint.org/docs/latest/rules/no-multi-assign)

Will do

Reply:
Oh dreamy, thanks for pointing me to it. I'll raise a separate PR to enable it (in case there's a couple multi-assign in the mix).